packages/build/src/core/report_metrics.ts

```diff
@@ -0,0 +1,31 @@
+import { closeClient, formatTags, InputStatsDOptions, startClient, validateStatsDOptions } from '../report/statsd.js'
+
+export type Metric = {
+  type: 'increment'
+  name: string
+  value: number
+  tags: Record<string, string | string[]>
+}
+
+export type Metrics = Array<Metric>
+
+/**
+ * Record number of functions build and differentiate between autogenerated and user generated.
+ * Sends to statsd daemon.
+ */
+export const reportMetrics = async function (statsdOpts: InputStatsDOptions, metrics: Metrics): Promise<void> {
+  if (!validateStatsDOptions(statsdOpts)) {
+    return
+  }
+
+  if (metrics.length === 0) return
```

Comment:
nit: We could group these together?

```suggestion
  if (!validateStatsDOptions(statsdOpts) || metrics.length === 0) {
    return
  }

```

Reply:
I like this as well! Will do